lib/OpenQA/Schema/ResultSet/Jobs.pm

```diff
@@ -118,6 +118,28 @@ sub latest_jobs {
     return @latest;
 }
 
+sub latest_jobs_multiple_groups {
+    my ($self, $until) = @_;
+
+    my @jobs = $self->search($until ? {t_created => {'<=' => $until}} : undef, {order_by => ['me.id DESC']});
+    my @latest;
+    my %seen;
+    foreach my $job (@jobs) {
+        my $groupid = $job->group_id;
+        my $test    = $job->TEST;
+        my $distri  = $job->DISTRI;
+        my $version = $job->VERSION;
+        my $build   = $job->BUILD;
+        my $flavor  = $job->FLAVOR;
+        my $arch    = $job->ARCH;
+        my $machine = $job->MACHINE // '';
+        my $key     = "$groupid-$distri-$version-$build-$test-$flavor-$arch-$machine";
+        next if $seen{$key}++;
+        push(@latest, $job);
+    }
+    return @latest;
+}
```

Comment:
that's good enough for a proof of concept but I would like this duplication to be removed before being able to approve